src/classes/TDTM_ObjectDataGateway.cls

```diff
@@ -91,9 +91,36 @@ public without sharing class TDTM_ObjectDataGateway implements TDTM_iTableDataGa
      */
     public static list<Trigger_Handler__c> listTH {
         get {
+            Boolean useAdv = false;
+            String apiNpspClass = '';
+
+            //Check if Advancement is installed and enabled.
+            try {
+                List<SObject> advSettings = Database.query('select Use_Advancement__c, API_NPSP_Class__c from Advancement_Setting__mdt');
+                useAdv = (Boolean)advSettings[0].get('Use_Advancement__c');
+                apiNpspClass = (String)advSettings[0].get('API_NPSP_Class__c');
+            } catch(Exception e) {
+                // Advancement not installed. Nothing to do. If the custom setting Adv_Settings doesn't exist, we'll use
+                // HEDA's TDTM config.
+            }
+
             if (listTH == null) {
```

Comment:
Here we say if the Trigger Handler is empty, then we try to use Adv version of Trigger Handler. But I think we also need to take care of when listTH is not empty. We still need Adv to overwrite the list of Trigger Handlers.